### DISCUSSION TITLE
Asynchronously read entry data

### DISCUSSION THREAD

**Discussion by sntran** [gildas-lormeau/zip.js#444] [Q&A]
I have a playground on [Deno Deploy](https://dash.deno.com/playground/unzip) with a demo at https://unzip.deno.dev.

Right now I have to `await entry.getData(writable)` in an async IIFE to avoid blocking before passing the `readable` to the Response. Otherwise, the response will finish early, throwing error of writable stream already being closed.

I would like to see if we can do that asynchronously so that the user can start receiving data immediately.

**Comment by gildas-lormeau** [comment]
Remove the `await` keyword :). You can use `.catch()` to catch errors if necessary.
